crates/sui-indexer-alt-jsonrpc/src/data/kv_loader.rs

```diff
@@ -116,4 +130,107 @@ impl KvLoader {
                 .transpose(),
         }
     }
+
+    pub(crate) async fn load_one_transaction(
+        &self,
+        digest: TransactionDigest,
+    ) -> Result<Option<TransactionContents>, Arc<Error>> {
+        let key = TransactionKey(digest);
+        match self {
+            Self::Bigtable(loader) => Ok(loader
+                .load_one(key)
+                .await?
+                .map(TransactionContents::Bigtable)),
+            Self::Pg(loader) => Ok(loader.load_one(key).await?.map(TransactionContents::Pg)),
+        }
+    }
+}
+
+impl TransactionContents {
+    pub(crate) fn data(&self) -> anyhow::Result<TransactionData> {
+        match self {
+            Self::Pg(stored) => {
+                let data: TransactionData =
+                    bcs::from_bytes(&stored.raw_transaction).map_err(|e| {
+                        anyhow::anyhow!("Failed to deserialize transaction data: {}", e)
+                    })?;
+
+                Ok(data)
+            }
+            Self::Bigtable(kv) => Ok(kv.transaction.data().transaction_data().clone()),
+        }
+    }
+
+    pub(crate) fn digest(&self) -> anyhow::Result<TransactionDigest> {
+        match self {
+            Self::Pg(stored) => {
+                let digest =
+                    TransactionDigest::try_from(stored.tx_digest.clone()).map_err(|e| {
+                        anyhow::anyhow!("Failed to deserialize transaction digest: {}", e)
+                    })?;
+
+                Ok(digest)
+            }
+            Self::Bigtable(kv) => Ok(*kv.transaction.digest()),
+        }
+    }
```

Comment:
Combining a couple of nits that apply to all these accessors:

- Use `Context` as mentioned above.
- You don't need to assign these to a variable.
- It would be nice to start with the simple case (the `Bigtable` case) so it doesn't get lost under the more complicated case.
- Pull the `Ok` out of the `match`

All together:

```rust
pub(crate) fn digest(&self) -> anyhow::Result<TransactionDigest> {
    Ok(match self {
        Self::Bigtable(kv) = *kv.transaction.digest(),
        Self::Postgres(stored) => TransactionDigest::try_from(stored.tx_digest.clone())
            .context("Failed to deserialize transaction digest")?
    })
}
```